images_of_bot5.0.py

```diff
@@ -696,40 +696,30 @@ def search_for_places(r):
 
 def swim(r, goodregex, postinto, getfromthese, submission, badregex=r"(\bbadregex\b)", badcaseregex=r"(\bBadCaseRegex\b)", smallsubredditblacklist={'blacklistedsubreddit'}):
         title = submission.title
-        if re.search(goodregex, title, flags=re.IGNORECASE) != None:
-            if re.search(badregex, title, flags=re.IGNORECASE) == None:
-                if re.search(badcaseregex, title,) == None:
-                    if str(submission.author).lower() not in globaluserblacklist:
-                        if str(submission.subreddit).lower() not in globalsubredditblacklist:
-                            if str(submission.subreddit).lower() not in smallsubredditblacklist:
-                                if submission.over_18 == False:
-                                    if re.search(urlregex, submission.url, flags=re.IGNORECASE) != None:
-                                        make_post(r, submission, postinto)
-        else:
-            if str(submission.subreddit).lower() in getfromthese:
-                if str(submission.author).lower() not in globaluserblacklist:
-                
-                    if submission.over_18 == False:
-                        if re.search(urlregex, submission.url, flags=re.IGNORECASE) != None:
-                            make_post(r, submission, postinto)
+        if (submission.over_18  or
+                submission.author.name.lower() in globaluserblacklist or
+                not re.search(urlregex, submission.url, flags=re.IGNORECASE)):
+            return
+        if ((submission.subreddit.display_name.lower() not in globalsubredditblacklist + smallsubredditblacklist and
+                re.search(goodregex, title, flags=re.IGNORECASE) and
+                not re.search(badregex, title, flags=re.IGNORECASE) and
+                not re.search(badcaseregex, title)) or
+                submission.subreddit.display_name.lower() in getfromthese:
```

Comment:
if val:
        do x
    elif val2:
        do x

can be simplified to

    if val or val2: do x

because in both cases, if the first val matches the interpreter short circuits to do x, if neither match the compiler doesn't do anything, and if the first fails, the second is checked. The order, however, must stay the same between both versions (and it is staying the same here)